queue.go

```diff
@@ -365,6 +365,10 @@ func (queue *redisQueue) ReceiveBatch(batchSize int, waitTime time.Duration) ([]
 		// redisErrIsNil may panic because the transport may reuse connections unexpectedly.
 		// instead, perform check in-line and don't panic if Redis has the problem.
 		if err := result.Err(); err != nil {
+			// Filter these two errors out to reduce log cruft.
+			if err == redis.Nil || strings.Contains(err.Error(), "i/o timeout") {
```

Comment:
A couple of comments:

1. If this is primarily being done to reduce log cruft, shouldn't the check and error log supression be done by the code that's calling this function?
2. Checking for a particular error type by string-matching on `err.Error()` can be OS-dependent and brittle.  There is usually a better way involving a type assertion.